---
title: "2_Alpha_diversities"
author: "Marly"
date: "2024-08-07"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

One has to consider the sequencing depth (how much of the taxa have been sampled) for each sample. If there is a large difference, then it is important to normalize the samples to equal sampling depth. First, we look at the sampling depth (no. of reads per sample).

Load packages

```{r}
library(microbiome) # data analysis and visualisation
library(phyloseq) # also the basis of data object. Data analysis and visualisation
library(microbiomeutilities) # some utility tools 
library(RColorBrewer) # nice color options
library(ggpubr) # publication quality figures, based on ggplot2
library(DT) # interactive tables in html and markdown
library(data.table) # alternative to data.frame
library(dplyr) # data handling
library(microbiome) # data analysis and visualisation
library(phyloseq) # also the basis of data object. Data analysis and visualisation
library(microbiomeutilities) # some utility tools 
library(RColorBrewer) # nice color options
library(ggpubr) # publication quality figures, based on ggplot2
library(DT) # interactive tables in html and markdown
library(data.table) # alternative to data.frame
library(dplyr) # data handling 
library(knitr)
library(ggplot2)
library(RSQLite)
library(DECIPHER)
library(DESeq2)
library(gridExtra)
library(ggnewscale) ##new_scale()
library(ggpubr) # ggarrange and ggerrorplot function
library(ggthemes)
library(grid) # rasterGrob()
library(vegan)
library(phyloseq)
library(png)
library(qgraph) # graph layout
library(ggsignif)
library(RColorBrewer)
library(tidyverse) # includes: ggplot2, dplyr, tidyr, readr, purr, tibble, stringr, forcats
library(knitr)
library(microbiomeAnalysis)
library(patchwork)

list_of_packages <- c("microbiomeAnalysis", "RColorBrewer", "microbiome", "CoDaSeq",
                      "compositions", "GGally", "dplyr", "tidyr", "ggbeeswarm",
                      "ALDEx2", "ggpubr", "kernlab", "cowplot")

lapply(list_of_packages, library, character.only = TRUE)
exp_color <- c("#7fc97f", "#beaed4", "#fdc086", "#A62800", "#03436A", "steelblue", "brown3", "grey50", "#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442" )

Palette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7","maroon","khaki4","coral4","blueviolet","chocolate3","gray48")




```

The data for this scr is stored as ps1.rds file in the R project phyobjects folder.

I will use the decontaminated phyloseq object from the Pre-processing.

After performing decontamination on the samples based on the frequency according to the DNA concentration across the entire dataset, I also conducted decontamination based on the collection. Through this process, I successfully removed a total of 40 contaminant taxa.

```{r}

#load the phylooseq objects

ps1 <- readRDS("/Users/u_erazo/Documents/LABbook/2024/Namibia_sequencing/analysis/phyobjects/Namibia_samples_decont.rds")

ps1
```


```{r}
summary(sample_sums(ps1))
```

```{r}
a <- plot_read_distribution(ps1 , groups = "sample_or_control", plot.type = "density") + theme_biome_utils()
a
```

```{r}
ggsave (".png",
        width=100,
        height=70,
        units="mm",
        scale=2)
```

```{r}
sample_dat_info <- meta(ps1)
sample_dat_info$sdt <- sample_sums(ps1)
```



As is evident there is a large difference in the number of reads. Minimum is 3 and maximum is 66186!! There is a 22062X difference!

We can plot the rarefaction curve for the observed ASVs in the entire data set. This is a way to check how has the richness captured in the sequencing effort.
```{r}
otu_tab <- t(abundances(ps1))
p <- vegan::rarecurve(otu_tab, 
                      step = 50, label = FALSE, 
                      sample = min(rowSums(otu_tab), 
                                   col = "blue", cex = 0.6))


```


# I want to check if there are differences in DNA concentration between the samples. This way, I can also see if maybe microbiome differences correlate with the amount of DNA in the samples.

```{r}

media_dnaconc <- sample_dat_info %>% 
group_by(sample_or_control) %>%  # group different replicates with treatments 
      summarise(median_replicates = median(`conc_16s__PCR`),  #calculate mean of area  of all replicates
                sd_area_replicates = sd(`conc_16s__PCR`))
    print(media_dnaconc)
    
```


```{r}
#plot median DNA conc
g <- ggplot(media_dnaconc, aes(x = sample_or_control, y = median_replicates)) +
  geom_boxplot( 
    aes(fill = sample_or_control),  # Agrega color = temperature
    width = 0.5, 
    size = 0.4,
    position = position_dodge(0.8),
    outlier.shape = NA, coef = 1.5, color = "black",  alpha = 0.5, width = 0.4
  ) 
   
g
```


```{r}
ggsave (".../ll.png",
        width=100,
        height=70,
        units="mm",
        scale=3)
```


The p-value of the test is 0.00103, which is less than the significance level alpha = 0.05. We can conclude that sdt and conc_16s__PCR are significantly correlated with a correlation coefficient of -0.17 and p-value of  0.00103 .


```{r}
ps_filter0 = prune_samples(sample_sums(ps1)>=500, ps1) 
ps_filter0 

ps_filter1<- ps_filter0 

#  ps_filter1 = prune_samples(sample_data(ps_filter0)$conc_16s__PCR >=3, ps_filter0)
# ps_filter1_data<- meta(ps_filter1)
``` 
```{r}
plot_read_distribution(ps_filter0 , groups = "sample_or_control", plot.type = "density") + theme_biome_utils()

```

```{r}
# ggsave ("....png",
#         width=100,
#         height=70,
#         units="mm",
#         scale=2)
```


The plot below is only for a quick and dirty check for reads per samples.
```{r}
 # quick check for sampling depth
barplot(sample_sums(ps_filter1), las =2)
```
 
Quick check taxa prevalence.
```{r}
# quick check taxa prevalence
tax_table(ps_filter1)

p.rar <- plot_taxa_prevalence(ps_filter1, "Phylum")

p.rar
```
 


```{r include=TRUE}
ps0.rar <- add_seqdepth(ps_filter1)

sample_data(ps0.rar) %>% data.frame() %>%
  mutate(idx = rownames(sample_data(ps0.rar))) %>% 
  ggplot(aes(x = Depth, fill = sample_or_control)) +
  geom_histogram(bins = 50) +
  geom_vline(xintercept = 500, col = "red") 
```
Following is the alpha diversity measurements Total OTUs observed, Inverse Simpson (effective number of OTUs), Shannon ("information"), and evenness (Shannon/log(total OTUs), is 1 if the OTUs are evenly distributed, 0 if not)

```{r}
ps_all <- ps0.rar
ps_all_alpha <- estimate_richness(ps_all, split = T,
                                measures = c("Observed", "Shannon", "InvSimpson")) 
ps_all_alpha <- merge(data.frame(sample_data(ps_all)), ps_all_alpha, by = 0)
```

```{r include=TRUE, fig.height=3, fig.width=7}

ps_all_alpha %>%

  
  mutate(Evenness = Shannon/log(Observed)) %>% 
  pivot_longer(cols = c(Observed, Shannon, InvSimpson, Evenness),
               names_to = "Diversity.index", values_to = "Value") %>%
  mutate(Diversity.index = factor(Diversity.index, 
                                  levels = c("Observed", "InvSimpson", "Shannon", "Evenness"))) %>%
  ggplot(aes(x = sample_or_control, y = Value)) +
  theme_bw(14) +
  geom_quasirandom() +

  stat_summary(fun = "median", col = "black", geom = "point") +
    # 
  theme(legend.position = "none")
```

Diversities

Non-phylogenetic diversities

Let us calculate diversity.
```{r}
hmp.div <- microbiome::alpha(ps0.rar, index = "all")
datatable(hmp.div)
```
