---
title: "Beta diversity metrics"
author: "Marly"
date: "2024-08-07"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```

Load packages and data
```{r}
# Load necessary libraries without repetition
library(microbiome) # data analysis and visualization
library(phyloseq) # basis of data object, data analysis and visualization
library(RColorBrewer) # nice color options
library(ggpubr) # publication quality figures, based on ggplot2
library(dplyr) # data handling  
library(tidyverse) # collection of R packages for data science
library(microbiomeAnalysis) # microbiome analysis
library(vegan) # tools for descriptive community ecology
library(ComplexHeatmap) # complex heatmaps
library(ggtext) # improved text rendering for ggplot2
library(ggraph) # graph visualization
library(DT) # data tables
library(corncob) # differential abundance testing
library(microViz) # microbiome data visualization
library(DECIPHER) # sequence analysis
library(ape) # phylogenetics and evolution
library(DESeq2) # differential expression analysis
library(ggplot2) # graphing package
library(plotly) # interactive web graphics
library(philr) # compositional data analysis
library(adespatial) # multiscale spatial analysis
library(devtools) # package development
library(microbiomeSeq) # data analysis and visualization
library(pander) # rendering R objects into Pandoc's markdown
library(ranacapa) # data analysis
library(grid) # data visualization support
library(gridExtra) # data visualization support
library(knitr) # dynamic report generation
library(png) # figure download
library(ggdendro) # dendrograms and tree plots using 'ggplot2'
library(microbiomeutilities) # utility tools

colors=c("#E4A365",
         "#00A1D5",
         "#BF6866")


```



Open the  phyloseq 
```{r}
ps <- readRDS("../phyloseq/Namibia_samples_decont.rds")

ps
```


```{r}
 rank_names(ps)
```




I will filter the phyloseq object to see differences between larvae and water at different temperatures

```{r}

ps.filt <- filter_physeq(ps, min_reads = 500, min_prop = 0.01, min_occurr = 0.0)
ps.filt <- add_taxa_info(ps.filt)

ps.filt
```


## PCA ordination

```{r}
myPal <- c()            
myPal["Elizabethkingia"] <-"#5B8FA8FF"
myPal["Asaia"] <- "#800000FF" 
myPal["Serratia"] <- "#FFB547FF" 
myPal["Enterobacter"] <- "#8A8B79FF" 
myPal["Other"] <-  "#ADB17DFF"
myPal["Rahnella"] <-  "#725663FF"
myPal["Chryseobacterium"] <- "#D49464FF" 
myPal["Yersinia"] <-  "#B1746FFF" 
myPal["Comamonas"] <-   "#9A5324FF" 
myPal["Undibacterium"] <-   "#C5B0D5"
myPal["Acinetobacter"] <-  "#350E20FF"
myPal["Masillia"] <-  "#767676FF"
myPal["Bosea"] <-  "#BCBD22"
myPal["Delfia"] <- "#0F425CFF"

```



```{r}
# Asegúrate de tener el paquete ggplot2 instalado y cargado
# install.packages("ggplot2")
library(ggplot2)

# Suponiendo que larvae.data es tu conjunto de datos y median_rep_end es la variable de días de pupación

ps%>%
 tax_transform(rank = "unique", trans = "identity") %>%
 dist_calc(dist = "bray") %>%
 ord_calc(method = "NMDS") %>% 
 ord_plot(
   axes = c(1, 2),
   colour = "sample_or_control",
   alpha = 1,
   size = 3.5
) +
 theme_bw(base_size = 14) 
 # scale_color_gradient(low = "lightblue", high = "darkblue") + # Cambia los colores a tu preferencia
 # scale_fill_gradient(low = "lightblue", high = "darkblue")

```


```{r}
ps_fix <- ps %>%
 tax_fix(
  min_length = 4,
  unknowns = c(""),
  sep = " ", anon_unique = TRUE,
  suffix_rank = "classified"
 )
```



```{r}

library(microViz) 

p <- ps_fix %>%
tax_filter( tax_level = "Genus") %>%
  tax_agg("Genus") %>%
  tax_transform("clr") %>%
  ord_calc("PCA")

# create plot
pca_plot <- p %>%
  ord_plot(
    axes = 2:3,
    plot_taxa = 1:5, colour = "sample_or_control", size = 3,
     tax_vec_length = 0.3, 
    tax_lab_style = tax_lab_style(max_angle = 90, aspect_ratio = 0.5, size = 4),
     auto_caption = 8
  ) +
  theme_bw(base_size = 16) +
  # theme(legend.position = "bottom") +
  coord_fixed(ratio = 0.5, clip = "off")  # makes r
  

# show plot
  pca_plot
```

```{r}
# 
# ggsave ("/.jpg",
#         width=100,
#         height=70,
#         units="mm",
#         scale=1.5)

```

